addons/l10n_ch/models/res_bank.py

```diff
@@ -42,30 +72,57 @@ def retrieve_acc_type(self, acc_number):
         """ Overridden method enabling the recognition of swiss postal bank
         account numbers.
         """
-        if _is_l10n_ch_postal(acc_number):
+        try:
+            validate_l10n_ch_postal(acc_number)
             return 'postal'
-        else:
+        except ValidationError:
             return super(ResPartnerBank, self).retrieve_acc_type(acc_number)
 
     @api.onchange('acc_number')
     def _onchange_set_l10n_ch_postal(self):
         if self.acc_type == 'iban':
             self.l10n_ch_postal = self._retrieve_l10n_ch_postal(self.sanitized_acc_number)
-        else:
-            self.l10n_ch_postal = self.sanitized_acc_number
+        elif self.acc_type == 'postal':
+            self.l10n_ch_postal = self.acc_number
```

Comment:
Is a postal code then a real postal code?   I thought we would take away the onchange all together.

Reply:
I thought also we should remove the onchange but you had that feature almost ready.

Reply:
2 formats are valids for postal account number:

9 digits : 100002345
formated : 10-234-5

The most used is the formatted version.

Reply:
We can replace this by a warning, when the user set a postal account number in acc_number we ask him to move it to l10n_ch_postal and set a description in acc_number.